irksome/stepper.py

```diff
@@ -265,3 +297,253 @@ def advance(self):
+                        sub = bc.function_space_index()
+                        comp = bc.function_space().component
+                        if comp is not None:  # check for sub-piece of vector-valued
+                            Vsp = V.sub(sub).sub(comp)
+                            for j in range(num_stages):
+                                gcur -= dt*btilde[j]*ws[num_fields*j+sub].sub(comp)
+                            try:
+                                gdat = assemble(interpolate(gcur-u0.sub(sub).sub(comp), Vsp))
+                                gmethod = lambda g, u: gdat.interpolate(g-u.sub(sub).sub(comp))
+                            except:  # noqa: E722
+                                gdat = project(gcur-u0.sub(sub).sub(comp), Vsp)
+                                gmethod = lambda g, u: gdat.project(g-u.sub(sub).sub(comp))
+                        else:
+                            Vsp = V.sub(sub)
+                            for j in range(num_stages):
+                                gcur -= dt*btilde[j]*ws[num_fields*j+sub]
+                            try:
+                                gdat = assemble(interpolate(gcur-u0.sub(sub), Vsp))
+                                gmethod = lambda g, u: gdat.interpolate(g-u.sub(sub))
+                            except:  # noqa: E722
+                                gdat = project(gcur-u0.sub(sub), Vsp)
+                                gmethod = lambda g, u: gdat.project(g-u.sub(sub))
+                    self.gstuff = (gdat, gcur, gmethod, Vsp)
+
+            embbc = []
+            gblah = []
+            for bc in bcs:
+                blah = EmbeddedBCData(bc, self.t, self.dt, num_fields, num_stages, btilde, V, ws, self.u0)
+                gdat, gcur, gmethod, gVsp = blah.gstuff
+                gblah.append((gdat, gcur, gmethod))
+                embbc.append(DirichletBC(gVsp, gdat, bc.sub_domain))
+            self.embbc = embbc
+            self.gblah = gblah
+
+    def _estimate_error(self):
+        """Assuming that the RK stages have been evaluated, estimates
+        the temporal truncation error by taking the norm of the
+        difference between the new solutions computed by the two
+        methods.  Typically will not be called by the end user."""
+        dtc = float(self.dt)
+        delb = self.delb
+        ws = self.ws
+        nf = self.num_fields
+        ns = self.num_stages
+        u0 = self.u0
+
+        # Initialize e to be gamma*h*f(old value of u)
+        error_func = Function(u0.function_space())
+        # Only do the hard stuff if gamma0 is not zero
+        if self.gamma0 != 0.0:
+            error_test = TestFunction(u0.function_space())
+            f_form = inner(error_func, error_test)*dx-self.gamma0*dtc*self.dtless_form
+            for gdat, gcur, gmethod in self.gblah:
+                gmethod(gcur, self.u0)
+            f_problem = NLVP(f_form, error_func, bcs=self.embbc)
+            solver_params = {
```

Comment:
Does this fail in parallel (are we defaulting the LU solver to PETSc, which is serial)?  And is there a way to override/select better options?  Looks like just a mass matrix, so probably CG with "something" could be made to work if we want to plumb in extra kwarg for these solver options.

Reply:
Good point - I had meant to do this, but forgotten.  Will plumb in options to select, but what do we have in mind as _good_ defaults here?

Reply:
Whatever firedrake's default linear solver is?  Or CG + ASM/ICC on blocks should be ok on a mass matrix?